src/appleseed/renderer/meta/tests/test_phasefunction.cpp

```diff
@@ -0,0 +1,337 @@
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Artem Bishev, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.renderer headers.
+#include "renderer/global/globaltypes.h"
+#include "renderer/kernel/intersection/intersector.h"
+#include "renderer/kernel/lighting/tracer.h"
+#include "renderer/kernel/rendering/rendererservices.h"
+#include "renderer/kernel/shading/oslshadergroupexec.h"
+#include "renderer/kernel/shading/shadingcontext.h"
+#include "renderer/kernel/texturing/texturecache.h"
+#include "renderer/kernel/texturing/texturestore.h"
+#include "renderer/modeling/entity/onframebeginrecorder.h"
+#include "renderer/modeling/input/scalarsource.h"
+#include "renderer/modeling/phasefunction/henyeyphasefunction.h"
+#include "renderer/modeling/phasefunction/phasefunction.h"
+#include "renderer/modeling/scene/containers.h"
+#include "renderer/modeling/scene/scene.h"
+#include "renderer/modeling/texture/texture.h"
+#include "renderer/utility/paramarray.h"
+#include "renderer/utility/testutils.h"
+
+// appleseed.foundation headers.
+#include "foundation/math/rng/mersennetwister.h"
+#include "foundation/math/sampling/mappings.h"
+#include "foundation/math/scalar.h"
+#include "foundation/math/vector.h"
+#include "foundation/utility/arena.h"
+#include "foundation/utility/autoreleaseptr.h"
+#include "foundation/utility/gnuplotfile.h"
+#include "foundation/utility/test.h"
+
+// Boost headers.
+#include "boost/bind.hpp"
+#include "boost/ref.hpp"
+
+// Standard headers.
+#include <vector>
+
+using namespace foundation;
+using namespace renderer;
+
+TEST_SUITE(Renderer_Modeling_PhaseFunction)
+{
+    struct Fixture
+      : public TestFixtureBase
+    {
+        // Number of MC samples to do integrations.
+        static const int NumberOfSamples = 200000;
```

Comment:
That's a lot of samples. How long do tests take with a debug build? We try to make unit tests fast enough that they (collectively) take less than a minute to run in debug.